### PR TITLE
Feature/document list and document preview

### PR DESCRIPTION
**Pull request by loggerz**
Implements #41

### CODE REVIEW

File: app/src/main/java/com/github/se/travelpouch/ui/authentication/SignIn.kt [hunk 1]
```diff
@@ -45,7 +45,7 @@ fun SignInScreen(navigationActions: NavigationActions) {
           onAuthComplete = { result ->
             Log.d("SignInScreen", "User signed in: ${result.user?.displayName}")
             Toast.makeText(context, "Login successful!", Toast.LENGTH_LONG).show()
-            navigationActions.navigateTo(Screen.GREETING)
+            navigationActions.navigateTo(Screen.TRAVEL_LIST)
```
Comment (appellet) [inline app/src/main/java/com/github/se/travelpouch/ui/authentication/SignIn.kt:48]: See my comment above in AddActivityScreenTest

File: app/src/main/java/com/github/se/travelpouch/ui/dashboard/AddActivityScreen.kt
Comment (appellet) [inline app/src/main/java/com/github/se/travelpouch/ui/dashboard/AddActivityScreen.kt:1]: See my comment above in AddActivityScreenTest 
This kind of fix didn't have to be done in your PR. You could have concentrated on your work and opened an exit for it. As it's easier to navigate and implement the rest of the project, we can keep it that way for now.

File: app/src/main/java/com/github/se/travelpouch/ui/documents/DocumentList.kt [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+          //
+          //                Button(onClick = {
+          //                    documentViewModel.getDocuments()
+          //                }) {
+          //                    Text(text = "Load docs", style =
+          // MaterialTheme.typography.bodyMedium)
+          //                }
+          //            }
+          //
+          //            Spacer(modifier = Modifier.height(4.dp))
```
Comment (appellet) [inline app/src/main/java/com/github/se/travelpouch/ui/documents/DocumentList.kt:144]: This comments need to be removed

File: app/src/main/java/com/github/se/travelpouch/ui/documents/DocumentList.kt [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                        onNavigateToDocumentPreview()
+                      })
+                }
+              }
+
+          //            // Location Name
+          //            Text(
+          //                text = travelContainer.location.name,
+          //                style = MaterialTheme.typography.bodySmall,
+          //                color = Color.Gray)
```
Comment (appellet) [inline app/src/main/java/com/github/se/travelpouch/ui/documents/DocumentList.kt:174]: This comments need to be removed

File: app/src/main/java/com/github/se/travelpouch/ui/home/Overview.kt [hunk 4]
```diff
@@ -77,7 +77,7 @@ fun TravelListScreen(
                 modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp).padding(pd)) {
                   items(travelList.size) { index ->
                     TravelItem(travelContainer = travelList[index]) {
-                      // Handle item click
+                      navigationActions.navigateTo(Screen.TRAVEL_ACTIVITIES)
```
Comment (appellet) [inline app/src/main/java/com/github/se/travelpouch/ui/home/Overview.kt:80]: See my comment above in AddActivityScreenTest

File: app/src/main/java/com/github/se/travelpouch/ui/navigation/NavigationActions.kt
Comment (appellet) [inline app/src/main/java/com/github/se/travelpouch/ui/navigation/NavigationActions.kt:1]: See my comment above in AddActivityScreenTest

File: app/src/test/java/com/github/se/travelpouch/ui/navigation/NavigationActionsTest.kt
Comment (appellet) [inline app/src/test/java/com/github/se/travelpouch/ui/navigation/NavigationActionsTest.kt:1]: See my comment above in AddActivityScreenTest

**Review by appellet — CHANGES_REQUESTED** [review]
The implementation of the document list is really good. 
However, a few changes need to be made. We should also avoid making changes to other views directly in the same PR, but rather create an exit for that. Some documentation needs to be added to certain files.

File: app/src/androidTest/java/com/github/se/travelpouch/ui/dashboard/AddActivityScreenTest.kt
[the remark this answers is not in the document]
Comment (loggerz) [reply app/src/androidTest/java/com/github/se/travelpouch/ui/dashboard/AddActivityScreenTest.kt:1]: Yes SCRUM master !

File: app/src/main/java/com/github/se/travelpouch/MainActivity.kt
[the remark this answers is not in the document]
Comment (loggerz) [reply app/src/main/java/com/github/se/travelpouch/MainActivity.kt:1]: Yes, although this one was clearly in scope in order to fully integrate my new views. 100% agree for the rest

File: app/src/main/java/com/github/se/travelpouch/ui/dashboard/AddActivityScreen.kt
Comment (loggerz) [reply app/src/main/java/com/github/se/travelpouch/ui/dashboard/AddActivityScreen.kt:1]: What is an exit ?

File: app/src/main/java/com/github/se/travelpouch/model/documents/DocumentRepository.kt [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        val documents =
+            task.result?.documents?.mapNotNull { document -> fromSnapshot(document) } ?: emptyList()
+        onSuccess(documents)
+      } else {
+        task.exception?.let { e ->
+          Log.e("DocumentRepositoryFirestore", "Error getting documents", e)
+          onFailure(e)
+        }
+      }
+    }
```
[the remark this answers is not in the document]
Comment (loggerz) [reply app/src/main/java/com/github/se/travelpouch/model/documents/DocumentRepository.kt:59]: At this point we might also do a real generic version to wrap firebase repositories. Let's discuss this in #60

**Review by appellet — APPROVED** [review]
Everything looks good to me. The requested changes have been made.

**Pull request merged**